### ISSUE TITLE
Fix failing CI tests: black formatting, test failures, and false positives

### ISSUE THREAD

**Issue by raifdmueller** [docToolchain/asciidoc-linter#25]
## Problem

The CI pipeline has several failing checks that need to be fixed:

### 1. Black Formatting (run_black)
Two files need reformatting:
- `asciidoc_linter/rules/whitespace_rules.py`
- `tests/rules/test_heading_rules.py`

### 2. Test Failures (run_tests)
Two tests are failing:
- `tests/rules/test_heading_rules.py::TestHeadingAttributesAndRoles::test_attributes_after_heading`
- `tests/rules/test_heading_rules.py::TestHeadingAttributesAndRoles::test_roles_before_heading`

### 3. Self-test False Positives (self_test_readme)
The linter reports false positives on README.adoc, claiming every line is a "Section title" that needs blank lines:
```
Results for README.adoc:
✗ README.adoc, line 1: Section title should be followed by a blank line
✗ README.adoc, line 2: Section title should be preceded by a blank line
...
```

This appears to be a bug in the whitespace rules.

## Acceptance Criteria
- [ ] All files pass `python -m black --check .`
- [ ] All tests pass with `python -m pytest`
- [ ] `asciidoc-linter README.adoc` completes without false positives
